### PR TITLE
QOL-9159 sync from upstream

### PR DESCRIPTION
**Pull request by ThrawnCA**
This should bring our code in line with upstream.

Theoretically we may be able to stop maintaining our fork and revert to using upstream.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -174,7 +170,12 @@ To install XLoader:
 
    Ensure ``datastore`` is also listed, to enable CKAN DataStore.
 
-6. If it is a production server, you'll want to store jobs info in a more
+6. Starting CKAN 2.10 you will need to set an API Token to be able to
+   execute jobs against the server::
+
+     ckanext.xloader.api_token = <your-CKAN-generated-API-Token>
```
Comment (duttonw) [inline README.rst:176]: this seems troublesome to auto configure.

File: ckanext/xloader/plugin.py [hunk 2]
```diff
@@ -46,131 +52,143 @@ class xloaderPlugin(plugins.SingletonPlugin):
...
+            context,
+            {
                 "id": resource.id,
-            }
+            },
         )
         self._submit_to_xloader(resource_dict)
 
     # IResourceController
+    if toolkit.check_ckan_version("2.10"):
```
Comment (duttonw) [inline ckanext/xloader/plugin.py:138]: nice pattern

File: dev-requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-responses==0.12.0
+responses==0.10.9
```
Comment (duttonw) [inline dev-requirements.txt:1]: down grade?

File: pip-requirements.txt [hunk 4]
```diff
@@ -1,8 +0,0 @@
-ckantoolkit==0.0.4
-pika>=1.1.0
-pyOpenSSL==18.0.0
-redis
```
Comment (duttonw) [inline pip-requirements.txt:4]: so is these already in other places?

**Review by duttonw — APPROVED** [review]

File: pip-requirements.txt (hunk 4 above)
Comment (ThrawnCA) [reply pip-requirements.txt:4]: I believe it's been restructured to no longer need them.

File: dev-requirements.txt (hunk 3 above)
Comment (ThrawnCA) [reply dev-requirements.txt:1]: Apparently so. They're just test dependencies, though.

File: README.rst (hunk 1 above)
Comment (ThrawnCA) [reply README.rst:176]: It also seems odd given that XLoader, as I recall, doesn't have to make API calls like DataPusher did. We'll have to look more into this before we upgrade to 2.10.

File: ckanext/xloader/jobs.py [hunk 5]
```diff
@@ -479,23 +478,41 @@ def update_resource(resource, patch_only=False):
...
+    This method provides backwards compatibility for CKAN 2.9 that
+    supported both methods and previous CKAN versions supporting
+    only API Keys.
+    """
+    user = None
+    if get_user_from_token:
+        user = get_user_from_token(api_key_or_token)
+    if not user:
+        user = model.Session.query(model.User).filter_by(
+            apikey=api_key_or_token
```
Comment (duttonw) [inline ckanext/xloader/jobs.py:502]: @ThrawnCA : I believe CKAN security is improving which is why this is now in place.

as well as on line 512 etc.

**Pull request merged**